scipy/stats/morestats.py

```diff
@@ -2784,7 +2844,16 @@ def wilcoxon(x, y=None, zero_method="wilcox", correction=False):
         r_plus += r_zero / 2.
         r_minus += r_zero / 2.
 
-    T = min(r_plus, r_minus)
+    # return min for two-sided test, but r_plus for one-sided test
+    # the literature is not consistent here
+    # r_plus is more informative since r_plus + r_minus = count*(count+1)/2,
```

Comment:
Is it always the case that ` r_plus + r_minus = count*(count+1)/2`?  Or only for the `wilcox` alternative?
I plugged in an example from `TestWilcoxon.test_wilcoxon_arg_type`
```
x = [1, 2, 3, 0, -1, 3, 1, 2, 1, 1, 2]
d = np.asarray(x)
r = scipy.stats.rankdata(abs(d))
r_plus = np.sum((d > 0) * r, axis=0)
r_minus = np.sum((d < 0) * r, axis=0)
r_zero = np.sum((d == 0) * r, axis=0)
(r_plus, r_minus, r_zero),  len(d)*(len(d)+1)/2

((61.0, 4.0, 1.0), 66.0)
```

Reply:
`wilcox`:
since you exclude zero differences before ranking, you have indeed that `r_plus + r_minus = count*(count+1)/2`

if `alternative == "pratt"`, you drop the ranks of zeros, so in that case, this does not hold

if `alternative == "zsplit"`, split the ranks of zeros to `r_plus`and `r_minus` and you get again that
`r_plus + r_minus = count*(count+1)/2`

Reply:
Can you update the code comments to reflect that this is only true for 2 of the 3 alternatives? Perhaps something like:
```
# r_plus is more informative since r_plus + r_minus = count*(count+1)/2,
# i.e. the sum of the ranks, so r_minus and the min can be inferred.
# (If alternative='pratt', then r_plus + r_minus = count*(count+1)/2 - r_zero.)
# [3] uses the r_plus for the one-sided test, keep min for two-sided test
# to keep backwards compatability.
```
That will make it less confusing for someone trying to understand/debug some particular example.

Reply:
yes, i updated the comment